elasticsearch_dsl/aggs.py

```diff
@@ -140,7 +140,12 @@ def __iter__(self) -> Iterable[str]:
         return iter(self.aggs)
 
     def _agg(
-        self, bucket: bool, name: str, agg_type: str, *args: Any, **params: Any
+        self,
+        bucket: bool,
+        name: str,
+        agg_type: Union[Dict[str, Any], Agg[_R], str],
```

Comment:
mypy was perfectly happy with my naive `agg_type: str` type, but pyright flagged an issue in one of the examples that pointed at this as the problem.